mlflow/__init__.py

```diff
@@ -46,22 +46,54 @@
 import mlflow.tracking as tracking  # noqa: E402
 
 # model flavors
-import mlflow.fastai as fastai  # noqa: E402
-import mlflow.gluon as gluon  # noqa: E402
-import mlflow.h2o as h2o  # noqa: E402
-import mlflow.keras as keras  # noqa: E402
-import mlflow.lightgbm as lightgbm  # noqa: E402
-import mlflow.mleap as mleap  # noqa: E402
-import mlflow.onnx as onnx  # noqa: E402
-import mlflow.pyfunc as pyfunc  # noqa: E402
-import mlflow.pytorch as pytorch  # noqa: E402
-import mlflow.sklearn as sklearn  # noqa: E402
-import mlflow.spacy as spacy  # noqa: E402
-import mlflow.spark as spark  # noqa: E402
-import mlflow.statsmodels as statsmodels  # noqa: E402
-import mlflow.tensorflow as tensorflow  # noqa: E402
-import mlflow.xgboost as xgboost  # noqa: E402
-import mlflow.shap as shap  # noqa: E402
+_model_flavors_supported = []
+try:
+    # pylint: disable=unused-import
+    import pandas  # noqa: E402
+    import numpy  # noqa: E402
+except ImportError as e:
+    warnings.warn(
+        "Model flavors could not be loaded due to missing numpy " "or pandas dependencies. %s." % e
+    )
```

Comment:
I don't think we need to emit a separate warning message here. Users of the skinny client shouldn't expect the MLflow Models component to work out of the box (given the documentation that we'll provide), and users of the full client shouldn't encounter this warning.

Accordingly, I think it's acceptable to try/catch `ImportError` for all of the MLflow Models imports. We have test coverage ensuring that the full client successfully imports all of MLflow's flavor modules: https://github.com/mlflow/mlflow/blob/032ec8841dad13492a6eb3c9acdf78803ed90f76/tests/test_flavors.py#L36.

Reply:
Can we add an inline comment explaining why we're conditionally importing these modules? I.e. the skinny client doesn't support them